components/slot-fill/slot.js

```diff
@@ -69,11 +67,10 @@ class Slot extends Component {
 				return cloneElement( child, { key: childKey } );
 			} );
 		} );
-
 		return (
-			<div ref={ this.bindNode }>
+			<SlotWrap ref={ this.bindNode }>
```

Comment:
@aduth - can we use `Fragment` here and omit `ref` part?

Reply:
@gziolo would `Fragment` replace the ability to pass the component?

Reply:
Yes, I don't think we want this to be a meaningful node. Currently it's used as the target for rendering portal children. 

https://github.com/WordPress/gutenberg/blob/5ebefc369209667e5d015f875b18dfffe3a5af0d/components/slot-fill/fill.js#L73

So, we probably need the `ref` still (since `createPortal` requires a DOM node target). While I agree #6839 is a problem, I don't think we want developers to be concerned with controlling this wrapper (since, again, it needn't exist as a meaningful node). 

Two things we could explore:

- The `bubbleVirtually = false` variation of a `Fill` may not require the wrapper, since it's not using `createPortal`
   - Though, I'd like to encourage the virtual variation, maybe eventually deprecate `bubbleVirtually = false` unless this wrapper business becomes too much an issue
- Assign the portal target node as the _parent_ from which the slot is rendered, and avoid the wrapper node.
   - Maybe assign `ref` to a temporary node, register slot with temporary node's parent, then remove temporary node?

Reply:
See #7231